src/App.tsx

```diff
@@ -1,61 +1,94 @@
+import { useState } from 'react';
 import './App.scss';
 
-// import usersFromServer from './api/users';
-// import todosFromServer from './api/todos';
+import usersFromServer from './api/users';
+import todosFromServer from './api/todos';
+import { TodoList } from './components/TodoList';
+
+export const App: React.FC = () => {
+  const [currentTitle, setCurrentTitle] = useState('');
+  const [currentUserId, setCurrentUserId] = useState(0);
+  const [todos, setTodos] = useState(todosFromServer);
+  const [wasSubmitted, setWasSubmitted] = useState(false);
+
+  const isFormValid = currentTitle.trim() && currentUserId;
+
+  const formReset = () => {
+    setCurrentTitle('');
+    setCurrentUserId(0);
+  };
+
+  const onSubmit = (event: React.FormEvent<HTMLFormElement>) => {
+    event.preventDefault();
+
+    if (isFormValid) {
+      const newTodo = {
+        id: todos.length > 0 ? Math.max(...todos.map(todo => todo.id)) + 1 : 1,
+        title: currentTitle,
+        userId: currentUserId,
+        completed: false,
+      };
+
+      setTodos([...todos, newTodo]);
+
+      formReset();
+    } else {
+      setWasSubmitted(true);
+    }
+  };
 
-export const App = () => {
   return (
     <div className="App">
       <h1>Add todo form</h1>
 
-      <form action="/api/todos" method="POST">
+      <form action="/api/todos" method="POST" onSubmit={onSubmit}>
         <div className="field">
-          <input type="text" data-cy="titleInput" />
-          <span className="error">Please enter a title</span>
+          <label className="lable" htmlFor="titleInput">
```

Comment:
There is a typo in the class name `lable`. It should be `label` to ensure proper styling and functionality.